examples/gallery/plot/points-categorial.py

```diff
@@ -0,0 +1,48 @@
+"""
+Color points by categories
+---------------------------
+The :meth:`pygmt.Figure.plot` method can be used to plot symbols which are
+color-coded by categories.
+"""
+
+import numpy as np
+import pandas as pd
+import pygmt
+
+# Load sample iris data, and convert 'species' column to categorical dtype
+df = pd.read_csv("https://github.com/mwaskom/seaborn-data/raw/master/iris.csv")
+df["species"] = df.species.astype(dtype="category")
+
+# Use pygmt.info to get region bounds (xmin, xmax, ymin, ymax)
+# The below example will return a numpy array like [2.  4.4 4.3 7.9]
+region = pygmt.info(
+    table=df[["sepal_width", "sepal_length"]],  # x and y columns
+    per_column=True,  # report output as a numpy array
+)
+
+# Make our 2D categorial scatter plot, coloring each of the 3 species differently
+fig = pygmt.Figure()
+
+# Generate basemap of 10cm x 10cm size
+fig.basemap(
+    region=region,
+    projection="X10c/10c",
+    frame=['xafg+l"Sepal Width"', 'yafg+l"Sepal Length"', "WSen"],
+)
+
+# Define colormap to use for three categories
+pygmt.makecpt(cmap="inferno", color_model="+c", series=(0, 3, 1))
+
+fig.plot(
+    x=df.sepal_width,  # Use one feature as x data input
+    y=df.sepal_length,  # Use another feature as y data input
+    sizes=df.petal_width
+    / df.petal_length,  # Vary each symbol size according to the ratio of the two remaining features
+    color=df.species.cat.codes.astype(int),  # Points colored by categorical number code
+    cmap=True,  # Use colormap created by makecpt
+    no_clip=True,  # Do not clip symbols that fall exactly on the map bounds
+    style="cc",  # Use circles as symbols with size in centimeter units
+    transparency=40,  # Set transparency level for all symbols to deal with overplotting
+)
```

Comment:
Can we add a colorbar?

Reply:
Perhaps not easy with GMT 6.1.1, but may be easier with GMT 6.2.0: https://github.com/GenericMappingTools/gmt/issues/4388 and https://github.com/GenericMappingTools/gmt/pull/4390

Reply:
I tried to add a colorbar from the beginning, however, as pointed out by @seisman currently it's not as easy as I expected. Should we add the example at first glance without the colorbar and adjust it after GMT 6.2.0 is out?

Reply:
Yes, I think we could add a note about the missing color bar and the possible improvement in the upcoming GMT 6.2.0.

Reply:
Added a comment on this.